### PR TITLE
Support Python 3.10

### PR DESCRIPTION
**Pull request by druzhinin-kirill**
Provide support for 3.10.

While here, fix the ci workflow to pick up Python versions properly.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -15,17 +15,18 @@ classifiers = [
...
+    "Programming Language :: Python :: 3.10",
     "Programming Language :: Python :: 3.11",
     "Programming Language :: Python :: 3.12",
     "Intended Audience :: Developers",
     "Operating System :: OS Independent",
 ]
 
 dependencies = [
-    "dbt-core>=1.5",
+    "dbt-core>=1.5, <1.8",
```
Comment (druzhinin-kirill) [inline pyproject.toml:26]: I fixed it temporarily to see if it resolves the issue on CI.

File: src/dbt_score/config.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 from dataclasses import dataclass, field, replace
 from pathlib import Path
 from typing import Any, Final
 
 from dbt_score.rule import RuleConfig
 
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
```
Comment (matthieucan) [inline src/dbt_score/config.py:14]: Is it a transitive dependency? We probably should have this in `pyproject.toml` for Python < 3.11?

Comment (jochemvandooren) [reply src/dbt_score/config.py:14]: I think we need something like:

`tomli = { version = "...", python = "<3.11" }` in pyproject indeed

Comment (druzhinin-kirill) [reply src/dbt_score/config.py:14]: Yes, good point 👍

File: pyproject.toml (hunk 1 above)
Comment (druzhinin-kirill) [reply pyproject.toml]: Okay, it did not reproduce on ci, we are safe to remove it 🆗

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -18,14 +18,14 @@ jobs:
...
-        python: ["3.11", "3.12"]
+        python: ["3.10", "3.11", "3.12"]
 
     steps:
       - uses: actions/checkout@v4
       - name: Set up PDM
         uses: pdm-project/setup-pdm@v4
         with:
-          python-version: "3.11"
+          python-version: ${{ matrix.python }}
```
Comment (matthieucan) [inline .github/workflows/ci.yml:28]: Good catch 👌

**Review by matthieucan — APPROVED** [review]

**Pull request merged**